src/aics-image-viewer/components/App/index.tsx

```diff
@@ -489,22 +500,33 @@ const App: React.FC<AppProps> = (props) => {
       options.rawArrayOptions = { data: rawData, metadata: rawDims };
     }
 
-    loader.current = await loadContext.createLoader(path, { ...options });
-
-    const aimg = await loader.current.createVolume(loadSpec, (v, channelIndex) => {
-      // NOTE: this callback runs *after* `onNewVolumeCreated` below, for every loaded channel
-      // TODO is this search by name necessary or will the `channelIndex` passed to the callback always match state?
-      const thisChannelSettings = getOneChannelSetting(v.imageInfo.channelNames[channelIndex]);
-      onChannelDataLoaded(v, thisChannelSettings!, channelIndex);
-    });
+    let aimg: Volume;
+    try {
+      loader.current = await loadContext.createLoader(path, { ...options });
+
+      aimg = await loader.current.createVolume(loadSpec, (v, channelIndex) => {
+        // NOTE: this callback runs *after* `onNewVolumeCreated` below, for every loaded channel
+        // TODO is this search by name necessary or will the `channelIndex` passed to the callback always match state?
+        const thisChannelSettings = getOneChannelSetting(v.imageInfo.channelNames[channelIndex]);
+        onChannelDataLoaded(v, thisChannelSettings!, channelIndex);
+      });
+    } catch (e) {
+      showError(e);
+      setSendingQueryRequest(false);
+      throw e;
+    }
 
     const channelNames = aimg.imageInfo.channelNames;
     const newChannelSettings = setChannelStateForNewImage(channelNames);
     setAllChannelsUnloaded(channelNames.length);
 
     // initiate loading only after setting up new channel settings,
     // in case the loader callback fires before the state is set
-    loader.current.loadVolumeData(aimg);
+    loader.current.loadVolumeData(aimg).catch((e) => {
+      showError(e);
+      setSendingQueryRequest(false);
+      throw e;
```

Comment:
Seems like we reuse this error handling multiple times (L196-198, L514-516 and L526-528), I'd recommend extracting this logic into a function, and having `setLoadErrorHandler` and other places call the same function. something like:
```
const handleError = (e) => {
showError(e);
setSendingQueryRequest(false);
} ;

loader.current.loadVolumeData(aimg).catch(handlerError);
```